api/src/trace.rs

```diff
@@ -63,3 +64,72 @@ pub enum ExecutionEvent {
         msg: String,
     },
 }
+
+impl From<ExecutionTrace> for InvocationTrace {
+    fn from(e_trace: ExecutionTrace) -> InvocationTrace {
+        let mut invocation_stack: Vec<InvocationTrace> = Vec::new();
+
+        for event in e_trace.clone().events.into_iter() {
```

Comment:
This seems to both move the ExecutionTrace _and_ clone it. Is it possible to avoid both? Can this impl be `From<&ExecutionTrace>` without too much pain (we can then delegate a `From<ExecutionTrace>` to that if convenient)?